packages/grid/data-grid/src/DataGrid.test.tsx

```diff
@@ -297,6 +297,50 @@ describe('<DataGrid />', () => {
           width: `${2 * parseInt(secondColumnWidthVal, 10)}px`,
         });
       });
+
+      it('should set the columns width so that if fills the remaining width when "checkboxSelection" is used and the columns have "flex" set', () => {
+        const rows = [
+          {
+            id: 1,
+            username: 'John Doe',
+            age: 30,
+          },
+        ];
+
+        const columns = [
+          {
+            field: 'id',
+            flex: 1,
+          },
+          {
+            field: 'name',
+            flex: 0.5,
+          },
+        ];
+
+        render(
+          <div style={{ width: 200, height: 300 }}>
+            <DataGrid columns={columns} rows={rows} checkboxSelection />
+          </div>,
+        );
+
+        const columnsHeader: HTMLElement | null = document.querySelector(
+          '.MuiDataGrid-colCellWrapper',
+        );
+        const allColumnsExceptCheckbox = document.querySelectorAll(
+          '[role="columnheader"]:not([aria-colindex="1"])',
+        );
+        const availableColumnsWidth = checkboxSelectionColDef.width
+          ? columnsHeader!.offsetWidth - checkboxSelectionColDef!.width
+          : columnsHeader!.offsetWidth;
+        let columnsWidth = 0;
+
+        allColumnsExceptCheckbox.forEach((col) => {
+          columnsWidth += (col as HTMLElement).offsetWidth;
+        });
+
+        expect(availableColumnsWidth).to.equal(columnsWidth);
```

Comment:
-2 to account for the border as done in XGrid.

```suggestion
        expect(Array.from(document.querySelectorAll('[role="columnheader"]')).reduce((width, item) => width + item.clientWidth, 0)).to.equal(200 - 2);
```

Reply:
Yeah I saw that but it was either hardcoding the 2 or do what I did and select the DOM elements 😅. I'm ok with this tho.